HIP/hip-874.md

```diff
@@ -0,0 +1,60 @@
+---
+hip: 874
+title: Enhancing Topic Metadata Accessibility in the Hedera REST API
+author: Michael Kantor (@kantorcodes), Ty Smith (@ty-swirldslabs)
+type: Standards Track
+category: Mirror
+status: Draft
+created: 2024-02-14
+requested-by: TierBot
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/pull/883
+---
+
+## Abstract:
+This HIP proposes an enhancement to the Hedera REST API to enable access to topic ID metadata, including the memo field and other relevant entity information. Currently, there is limited direct access to topic metadata through the REST API, which can hinder the development of applications that rely on this data for functionality or auditing purposes. This proposal aims to streamline access to topic metadata, reducing the need for workarounds and enhancing the capabilities of applications built on Hedera.
+
+## Motivation:
+The ability to access topic ID metadata, including the memo field, submit key, and deleted field directly through the REST API is crucial for a wide range of applications, including messaging, event tracking, and audit-ability functions. Currently, developers may need to employ indirect methods or multiple API calls to retrieve this information, leading to inefficiencies and potential inaccuracies. By providing direct access to this metadata, the proposal seeks to enhance developer experience, reduce computational load, and foster the development of more feature-rich applications on the Hedera platform.
+
+## Specification:
+The proposed enhancement involves extending the REST API to include a new endpoint to return topic ID metadata, including the memo field and other relevant entity information.
+
+### Route:
+/api/v1/topics/{topic id}
+
+Current Response Structure returns:
+```json
+{"_status":{"messages":[{"message":"Not found"}]}}
+```
+
+Proposed Response Structure:
+```json
+{
+  "topic_id": "0.0.123456",
+  "memo": "[Topic Memo]",
+  "admin_key": {
+      "_type": "ED25519",
+      "key": "bfc9d4a344389324119eed485eb3a875b1e5a54bd189d2fd5ad88ce1d3473dee"
+  },
+  "deleted": false,
+  "submit_key": {
+      "_type": "ED25519",
+      "key": "bfc9d4a344389324119eed485eb3a875b1e5a54bd189d2fd5ad88ce1d3473dee"
+  },
+  "auto_renew_period": 7890000,
+  "auto_renew_account_id": "0.0.654321",
+  "updated_timestamp" : "1707953087.386834512",
```

Comment:
Our standard is to use a timestamp range. This leaves open the possibility to support a `timestamp` query parameter for historical views of the topic.

```suggestion
  "timestamp" : {
    "from: "1707953087.386834512",
    "to": null
  }
```